legacy-contracts/locked-asset-distribution/src/lib.rs

```diff
@@ -42,15 +42,20 @@ pub trait Distribution {
     }
 
     #[endpoint(clearCommunityDistributionList)]
-    fn clear_community_distribution_list(&self, n: u64) -> usize {
-        for _ in 0..n {
-            self.community_distribution_list().pop_front();
+    fn clear_community_distribution_list(&self, n: u64) -> (u64, usize) {
+        let mut counter = 0;
```

Comment:
Not really mandatory but would be good to have 2 extra things if we have polished the code.

1. rename `n` with something more suggestive, like `elements_to_delete`
2. Add a require that the passed variable (whatever the rename) is > 0. Right now, if you pass 0, it will still remove 1 element.